### DISCUSSION TITLE
RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.

### DISCUSSION THREAD

**Discussion by pingcaoai** [mrdbourke/pytorch-deep-learning#54] [Q&A] [answered]
**Hi all!
I was running 01_pytorch_workflow.ipynb, and encountered this issue, can anybody help me?**
```
# Plot the loss curves
plt.plot(epoch_count, train_loss_values, label="Train loss")
plt.plot(epoch_count, test_loss_values, label="Test loss")
plt.title("Training and test loss curves")
plt.ylabel("Loss")
plt.xlabel("Epochs")
plt.legend();
```

![Screen Shot 2022-05-09 at 15 55 55](https://user-images.githubusercontent.com/54368185/167425938-3fcfa72f-1925-43e2-ad17-29daf2e57a80.png)

**Comment by mrdbourke** [comment, marked as answer]
Hi @creatorcao,

Thank you for the question, it seems there was an update to PyTorch causing this.

In the cell above the one you're trying to run, you can change the code to:

```python
# Print out what's happening
      if epoch % 10 == 0:
            epoch_count.append(epoch)
            train_loss_values.append(loss.detach().numpy()) # New
            test_loss_values.append(test_loss.detach().numpy()) # New
            print(f"Epoch: {epoch} | MAE Train Loss: {loss} | MAE Test Loss: {test_loss} ")
```

Notice the use of `loss.detach().numpy()` to remove the need for gradient tracking when appending the loss values.

Without this, the code will error.

I've updated the notebook to reflect this.
